src/main/java/org/apache/flink/benchmark/ContinuousFileReaderOperatorBenchmark.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.api.common.io.FileInputFormat;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.io.RowCsvInputFormat;
+import org.apache.flink.api.java.io.TextInputFormat;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.formats.avro.AvroInputFormat;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.util.Preconditions;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+/**
+ * Measures simple file reading and, optionally, parsing operation.
+ * Defaults to reading 1000 records from: <ol>
+ *     <li>single userdata.avro file</li>
+ *     <li>10 text files (100 lines each)</li>
+ *     <li>100 text (smaller) files (10 lines each)</li></ol>
+ * By default it runs with DOP 1 and then 2.
+ * <p>
+ * Parameters: <ul>
+ *     <li>path - folder or file; if not absolute, it's assumed relative to src/main/resources </li>
+ *     <li>fmt - format hint (optional: if not given it is inferred from file extension or starting of folder name)</li>
+ *     <li>dop - degree of parallelism.</li>
+ * </ul>
+ */
+// bash script to generate 10 files with 100 random base64 strings each:
+// for f in $(seq 1 10); do for l in $(seq 1 100); do dd if=/dev/random count=2 | base64 >> src/main/resources/txt-10-100-1024/$f; done; done
+@OperationsPerInvocation(value = ContinuousFileReaderOperatorBenchmark.RECORDS_PER_INVOCATION)
+@Fork(value = 1)
```

Comment:
Please use the default settings defined in the `BenchmarkBase`. `Fork(value = 1)` can introduce instabilities that might not be easily detectable, as different JVM instances can optimise could differently (one fork can be consistently faster while other consistently slower).

To offset the higher number of benchmarking runs, I would think about reducing number of tested parameters combination. Do we need to test with different formats? Or with different parallelism?